src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends Figure {
+    private static final int FIGURES_NUMBER = 5;
+    private static final int MAX_RANDOM_NUMBER = 20;
+    private static final int DEFAULT_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        int indexOfFigures = new Random().nextInt(FIGURES_NUMBER);
+
+        ColorSupplier colorSupplier = new ColorSupplier();
+
+        switch (indexOfFigures) {
+            case 1: {
+                Square square = new Square();
+                square.setColor(colorSupplier.getRandomColor());
+                square.setSide(new Random().nextInt(MAX_RANDOM_NUMBER));
+                return square;
+            }
+            case 2: {
+                Rectangle rectangle = new Rectangle();
+                rectangle.setColor(colorSupplier.getRandomColor());
+                rectangle.setFirstSide(new Random().nextInt(MAX_RANDOM_NUMBER));
+                rectangle.setSecondSide(new Random().nextInt(MAX_RANDOM_NUMBER));
+                return rectangle;
+            }
+            case 3: {
+                RightTriangle rightTriangle = new RightTriangle();
+                rightTriangle.setColor(colorSupplier.getRandomColor());
+                rightTriangle.setFirstLeg(new Random().nextInt(MAX_RANDOM_NUMBER));
+                rightTriangle.setSecondLeg(new Random().nextInt(MAX_RANDOM_NUMBER));
+                return rightTriangle;
+            }
+            case 4: {
```

Comment:
Lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)